src/rgw/motr/gc/gc.cc

```diff
@@ -136,6 +145,24 @@ void MotrGC::initialize() {
   // set random starting index for enqueue of delete requests
   enqueue_index = \
     ceph::util::generate_random_number(0, max_indices - 1);
+  // Init KV lock provider
+  std::unique_ptr<MotrLockProvider> kv_lock_provider =
+      std::make_unique<MotrKVLockProvider>();
+  if (rc == 0) {
```

Comment:
This will be always false as global rc is never updated.
Please remove the rc declaration inside for loop and use the global one.

Reply:
done